framework/include/utils/MooseMeshUtils.h

```diff
@@ -289,4 +289,21 @@ void makeOrderedNodeList(std::vector<std::pair<dof_id_type, dof_id_type>> & node
                          std::vector<dof_id_type> & elem_id_list,
                          std::vector<dof_id_type> & ordered_node_list,
                          std::vector<dof_id_type> & ordered_elem_id_list);
+
+/**
+ * Converts a given name (BoundaryName or SubdomainName) into a corresponding ID (BoundaryID or
+ * SubdomainID) and performs bounds checking to ensure that overflow doesn't happen.
+ * @param name Name that is to be converted into an ID.
+ * @return ID type corresponding to the type of name.
+ */
+template <typename T, typename Q>
+Q
+getIDFromName(const T & name)
+{
+  const auto id = stoi(name);
```

Comment:
this will always be an int, so if Q is unsigned int we have a problem

maybe stoll ? but we could just use convert in mooseUtils with a big long int type?

Reply:
or maybe just use >> like we used to?

Reply:
See current implementation, which uses stoll.